crates/oxc_linter/src/rules/no_new_symbol.rs

```diff
@@ -0,0 +1,73 @@
+use oxc_ast::{ast::Expression, AstKind, Span};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-new-symbol): Disallow new operators with the Symbol object")]
+#[diagnostic(
+    severity(warning),
+    help(
+        "Symbol is not intended to be used with the new operator, but to be called as a function. Consider removing the new operator."
+    )
+)]
+struct NoNewSymbolDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoNewSymbol;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow new operators with the Symbol object
+    ///
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Symbol is not intended to be used with the new operator, but to be called as a function.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// var foo = new Symbol('foo');
+    /// ```
+    NoNewSymbol,
+    correctness
+);
+
+impl Rule for NoNewSymbol {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::NewExpression(expr) = node.get().kind()
+            && let Expression::Identifier(ident) = &expr.callee
+            && ident.name == "Symbol"
+            && ctx.scope(node).unresolved_references.get("Symbol").is_some()
+        {
+            let start = expr.span.start;
+            let end = start + 3;
+            ctx.diagnostic(NoNewSymbolDiagnostic(Span { start, end }));
```

Comment:
Nit: use the constructor `Span::new`, I noticed this when I added other fields to `Span`.